boa/src/syntax/parser/expression/mod.rs

```diff
@@ -36,12 +72,10 @@ macro_rules! expression { ($name:ident, $lower:ident, [$( $op:path ),*], [$( $lo
             while let Some(tok) = cursor.peek(0) {
                 match tok.kind {
                     TokenKind::Punctuator(op) if $( op == $op )||* => {
-                        let _ = cursor.next().expect("token disappeared");
-                        lhs = Node::bin_op(
-                            op.as_binop().expect("could not get binary operation"),
-                            lhs,
-                            $lower::new($( self.$low_param ),*).parse(cursor)?
-                        )
+                        lhs = self.build_op(op, lhs, cursor)?;
```

Comment:
I understand the idea of adding the helper function, but this adds lots of code to optimise, with pretty complex generics, that might slow things down when compiling, and it's only two statements. Would it make sense duplicating the code in the two branches of the `match`?

Reply:
I have decided to remove the helper - you're right, it's too complicated for just two statements, haha.